tests/lib/rules/no-extra-boolean-cast.js

```diff
@@ -21,17 +21,38 @@ const ruleTester = new RuleTester();
 ruleTester.run("no-extra-boolean-cast", rule, {
 
     valid: [
-        "var foo = !!bar;",
-        "function foo() { return !!bar; }",
-        "var foo = bar() ? !!baz : !!bat",
-        "for(!!foo;;) {}",
-        "for(;; !!foo) {}",
-        "var foo = Boolean(bar);",
-        "function foo() { return Boolean(bar); }",
-        "var foo = bar() ? Boolean(baz) : Boolean(bat)",
-        "for(Boolean(foo);;) {}",
-        "for(;; Boolean(foo)) {}",
-        "if (new Boolean(foo)) {}"
+
+
+        /*
+         * "var foo = !!bar;",
+         * "function foo() { return !!bar; }",
+         * "var foo = bar() ? !!baz : !!bat",
+         * "for(!!foo;;) {}",
+         * "for(;; !!foo) {}",
+         * "var foo = Boolean(bar);",
+         * "function foo() { return Boolean(bar); }",
+         * "var foo = bar() ? Boolean(baz) : Boolean(bat)",
+         * "for(Boolean(foo);;) {}",
+         * "for(;; Boolean(foo)) {}",
+         * "if (new Boolean(foo)) {}",
+         * "var foo = bar || !!baz",
+         * "var foo = bar && !!baz",
+         * "var foo = bar || (baz && !!bat)",
+         * "function foo() { return (!!bar || baz); }",
+         * "var foo = bar() ? (!!baz && bat) : (!!bat && qux)",
+         * "for(!!(foo && bar);;) {}",
+         * "for(;; !!(foo || bar)) {}",
+         * "var foo = Boolean(bar) || baz;",
+         * "var foo = bar || Boolean(baz);",
+         * "var foo = Boolean(bar) || Boolean(baz);",
+         * "function foo() { return (Boolean(bar) || baz); }",
+         * "var foo = bar() ? Boolean(baz) || bat : Boolean(bat)",
+         * "for(Boolean(foo) || bar;;) {}",
+         * "for(;; Boolean(foo) || bar) {}",
+         * "if (new Boolean(foo) || bar) {}"
```

Comment:
These tests that show that the rule won't mistakenly report some logical operands should probably have `{ enforceForLogicalOperands: true }`.

Also, there should be a set of tests where the rule would indeed report logical operands if the option was enabled, but it isn't enabled.